sdk/core/core-lro/test/utils/testPoller.ts

```diff
@@ -0,0 +1,45 @@
+import { delay, RequestOptionsBase } from "@azure/core-http";
+import { Poller, PollOperationState } from "../../src";
+import { TestServiceClient } from "./testServiceClient";
+import { makeOperation, TestOperation, TestOperationProperties } from "./testOperation";
+
+export class TestPoller extends Poller<TestOperationProperties, string> {
+  public intervalInMs: number;
+
+  constructor(
+    client: TestServiceClient,
+    manual: boolean = false,
+    intervalInMs: number = 10,
+    requestOptions?: RequestOptionsBase,
+    baseOperation?: TestOperation
+  ) {
+    let state: PollOperationState = {};
+    let properties: TestOperationProperties | undefined = undefined;
+
+    if (baseOperation) {
+      state = baseOperation.state;
+      properties = baseOperation.properties;
+    }
+
+    const operation = makeOperation(state, {
+      ...properties,
+      client,
+      requestOptions
+    });
+
+    super(operation, manual);
+
+    this.intervalInMs = intervalInMs;
+  }
+
+  async delay(): Promise<void> {
+    return delay(this.intervalInMs);
+  }
+
+  async getResult(): Promise<string> {
```

Comment:
when getResult is called and the operation isn't in a final state, it should be able to return undefined, right?

Reply:
oh ok ok. I was thinking of it throwing. I'll fix it!

Reply:
Ah... that could work too, though we'd have to doc it.

Reply:
I'll add | undefined!